eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -430,6 +464,8 @@ object Peer {
     def apply(uri: NodeURI): Connect = new Connect(uri.nodeId, Some(uri.address))
   }
 
+  case class SendMessage(nodeId: PublicKey, message: LightningMessage)
```

Comment:
I'm not sure that this belongs to the `Peer`, this is more of a `Switchboard` message (since you want to potentially trigger opening a new connection if we don't have one with the target).

I'm also not sure it should be that generic (containing any `LightningMessage`).
Let's start with just onion messages (as it's the only case for which we're using it) and later we can make it more generic if needed.

Reply:
Done